database_size/models/ir_model_size.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 2025 Opener B.V. <https://opener.amsterdam>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from odoo import api, fields, models
+
+
+class IrModelSize(models.Model):
+    _name = "ir.model.size"
+    _description = "Disk space usage per model"
+    _order = "measurement_date desc, total_model_size desc"
+    _rec_name = "model_id"
+    _sql_constraints = [
+        (
+            "uniq_model_id_measurement_date",
+            "unique(model_id, measurement_date)",
+            "There is already a measurement for this model on the given date",
+        ),
+    ]
+    model_id = fields.Many2one(
+        comodel_name="ir.model",
+        index=True,
+        ondelete="cascade",
+        required=True,
+    )
```

Comment:
The current design dosn't allow to uninstall modules.
if you uninstall modules, the previous data will be dropped, which is not the expected behaviour.

I think we can replace by "model_name" here. So if module are uninstalled, data are conserved.

Reply:
Because you loose the size of the database in the past.

Reply:
I agree with @legalsylvain. It would be weird for the database size to shrink retroactively after uninstalling some module. The code is now updated accordingly.